File: src/model/coupouns.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

/**
 *
 * @author LENOVO
 */
public class coupouns {
    private int id_coupouns;
   private String text_coupouns;
   private String code_coupouns;
   private String date_1;
      private String date_2;
      private int id_promo1;
      private int duree_coupouns;

   

    public coupouns() {
    }

    public coupouns(int id_coupouns, String text_coupouns, String code_coupouns, String date_1, String date_2, int id_promo1, int duree_coupouns) {
        this.id_coupouns = id_coupouns;
        this.text_coupouns = text_coupouns;
        this.code_coupouns = code_coupouns;
        this.date_1 = date_1;
        this.date_2 = date_2;
        this.id_promo1 = id_promo1;
        this.duree_coupouns = duree_coupouns;
    }

    public coupouns(String text_coupouns, String code_coupouns, String date_1, String date_2, int id_promo1, int duree_coupouns) {
        this.text_coupouns = text_coupouns;
        this.code_coupouns = code_coupouns;
        this.date_1 = date_1;
        this.date_2 = date_2;
        this.id_promo1 = id_promo1;
        this.duree_coupouns = duree_coupouns;
    }
    

    public int getDuree_coupouns() {
        return duree_coupouns;
    }

    public void setDuree_coupouns(int duree_coupouns) {
        this.duree_coupouns = duree_coupouns;
    }

   

    public String getDate_1() {
        return date_1;
    }

    public void setDate_1(String date_1) {
        this.date_1 = date_1;
    }

    public String getDate_2() {
        return date_2;
    }

    public void setDate_2(String date_2) {
        this.date_2 = date_2;
    }

    public int getId_promo1() {
        return id_promo1;
    }

    public void setId_promo1(int id_promo1) {
        this.id_promo1 = id_promo1;
    }

 
    public int getId_coupouns() {
        return id_coupouns;
    }

    public void setId_coupouns(int id_coupouns) {
        this.id_coupouns = id_coupouns;
    }

    public String getText_coupouns() {
        return text_coupouns;
    }

    public void setText_coupouns(String text_coupouns) {
        this.text_coupouns = text_coupouns;
    }

    public String getCode_coupouns() {
        return code_coupouns;
    }

    public void setCode_coupouns(String code_coupouns) {
        this.code_coupouns = code_coupouns;
    }

    @Override
    public String toString() {
        return "coupouns{" + "id_coupouns=" + id_coupouns + ", text_coupouns=" + text_coupouns + ", code_coupouns=" + code_coupouns + ", date_1=" + date_1 + ", date_2=" + date_2 + ", id_promo1=" + id_promo1 + ", duree_coupouns=" + duree_coupouns + '}';
    }

     
    public int compareTo(coupouns c) {
        return Long.compare(this.id_coupouns, c.id_coupouns);
    }

  
   
}

File: src/Entity/Coupouns.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * Coupouns
 *
 * @ORM\Table(name="coupouns")
 * @ORM\Entity
 */
class Coupouns
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_coupouns", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idCoupouns;

    /**
     * @return int
     */
    public function getIdCoupouns(): int
    {
        return $this->idCoupouns;
    }

    /**
     * @param int $idCoupouns
     * @return Coupouns
     */
    public function setIdCoupouns(int $idCoupouns): Coupouns
    {
        $this->idCoupouns = $idCoupouns;
        return $this;
    }

    /**
     * @var string
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=15" )
     *
     * @ORM\Column(name="text_coupouns", type="string", length=200, nullable=false)
     */
    private $textCoupouns;

    /**
     * @var string
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=15" )
     *
     * @ORM\Column(name="code_coupouns", type="string", length=200, nullable=false)
     */
    private $codeCoupouns;

    /**
     * @var \DateTime
     * @Assert\NotBlank(message="description  doit etre non vide")
     *
     * @ORM\Column(name="date_offre", type="date", nullable=false)
     */
    private $date1;

    /**
     * @var \DateTime
     * @Assert\NotBlank(message="description  doit etre non vide")
     *
     * @ORM\Column(name="date_2", type="date", nullable=false)
     */
    private $date2;

    /**
     * @var int
     * @Assert\NotBlank(message="description  doit etre non vide")
     *
     * @ORM\Column(name="id_promo1", type="integer", nullable=false)
     */
    private $idPromo1;

    /**
     * @var int
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @ORM\Column(name="duree_coupouns", type="integer", nullable=false)
     */
    private $dureeCoupouns;

    /**
     * @return string
     */
    public function getTextCoupouns(): ?string
    {
        return $this->textCoupouns;
    }

    /**
     * @param string $textCoupouns
     * @return Coupouns
     */
    public function setTextCoupouns(string $textCoupouns): Coupouns
    {
        $this->textCoupouns = $textCoupouns;
        return $this;
    }

    /**
     * @return string
     */
    public function getCodeCoupouns(): ?string
    {
        return $this->codeCoupouns;
    }
    public function __toString() {
        return $this->codeCoupouns;
    }
    /**
     * @param string $codeCoupouns
     * @return Coupouns
     */
    public function setCodeCoupouns(string $codeCoupouns): Coupouns
    {
        $this->codeCoupouns = $codeCoupouns;
        return $this;
    }

    /**
     * @return \DateTime
     */
    public function getDate1(): ?\DateTime
    {
        return $this->date1;
    }

    /**
     * @param \DateTime $date1
     * @return Coupouns
     */
    public function setDate1(string $date1): Coupouns
    {
        $this->date1 = $date1;
        return $this;
    }

    /**
     * @return \DateTime
     */
    public function getDate2(): ?\DateTime
    {
        return $this->date2;
    }

    /**
     * @param \DateTime $date2
     * @return Coupouns
     */
    public function setDate2(string $date2): Coupouns
    {
        $this->date2 = $date2;
        return $this;
    }

    /**
     * @return int
     */
    public function getIdPromo1(): ?int
    {
        return $this->idPromo1;
    }

    /**
     * @param int $idPromo1
     * @return Coupouns
     */
    public function setIdPromo1(int $idPromo1): Coupouns
    {
        $this->idPromo1 = $idPromo1;
        return $this;
    }

    /**
     * @return int
     */
    public function getDureeCoupouns(): ?int
    {
        return $this->dureeCoupouns;
    }

    /**
     * @param int $dureeCoupouns
     * @return Coupouns
     */
    public function setDureeCoupouns(int $dureeCoupouns): Coupouns
    {
        $this->dureeCoupouns = $dureeCoupouns;
        return $this;
    }



}
